src/lib/blueprint.ts

```diff
@@ -1388,6 +1415,33 @@ const createProperty = (
   }
 }
 
+const validatePropertyGroup = (
+  propertyGroup: string,
+  propertyName: string,
+  parentPaths: string[],
+): void => {
+  if (propertyGroup.length === 0) return
+
+  const resourceName = parentPaths.at(0)
+  if (resourceName == null) {
+    throw new Error(
+      `Missing resource name for property "${propertyName}" in ${parentPaths.join('.')}`,
+    )
+  }
+
+  const groups = resourcePropertyGroups[resourceName] ?? {}
+  const validGroupKeys = Object.keys(groups)
+
+  if (validGroupKeys.length === 0) return
```

Comment:
If there are no `validGroupKeys` but `propertyGroup` is defined, this is an error right?
```suggestion
```

Reply:
Currently, resources don't define property groups, and we likely won't add them all at once - it will happen incrementally. So we can't throw an error here when a resource doesn't define property groups, right?

Reply:
af5109f3dd40b86298ba5531f0eac7d73d2b9535